### PR TITLE
AP-1753 Whitelist IP ranges that can access admin pages

### PR DESCRIPTION
**Pull request by stephenrichards**
## Whitelist IP ranges that can access admin pages

[Link to story](https://dsdmoj.atlassian.net/browse/AP-1753)

- created encrypted file `config/ipaddrs.crypt.yml` to list ranges that can access admin pages
- Created `AdminBaseController` that has a before action to call `AuthorizedIpRanges` to check that the request ip is in one of those ranges
- Derived all admin controllers from the `AdminBaseController`

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [ ] Tests and rubocop should be passing: `bundle exec rake`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [ ] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [ ] The PR description should say what you changed and why, with a link to the JIRA story.
- [ ] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -100,6 +100,7 @@ group :development, :test do
   gem 'byebug', platforms: %i[mri mingw x64_mingw]
   gem 'dotenv-rails', '>= 2.7.5'
   gem 'erb_lint', require: false
+  gem 'htmlentities'
```
Comment (mpw5) [inline Gemfile:103]: This doesn't seem to be related to this PR. Has it crept in from another change?

File: spec/requests/citizens/check_answers_spec.rb [hunk 2]
```diff
@@ -60,7 +60,7 @@
 
     it 'displays the name of the firm' do
       subject
-      expect(response.body).to include(firm.name)
+      expect(response.body).to include(HTMLEntities.new.encode(firm.name))
```
Comment (mpw5) [inline spec/requests/citizens/check_answers_spec.rb:63]: This doesn't seem to be related to this PR. Has it crept in from another change?

**Review by mpw5 — COMMENTED** [review]
Looks good, but there appear to be some changes in the PR which don't relate to this change. Have then been added by mistake?

File: Gemfile (hunk 1 above)
Comment (stephenrichards) [reply Gemfile:103]: No - I included it because I was getting spec failures when Faker generates a name like "O'Keefe" - so this gem enables us to test against those names when they crop up as well.

File: spec/requests/citizens/check_answers_spec.rb (hunk 2 above)
Comment (stephenrichards) [reply spec/requests/citizens/check_answers_spec.rb:63]: See above

**Review by mpw5 — APPROVED** [review]
LGTM
